addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/data/Thermostat.java

```diff
@@ -0,0 +1,406 @@
+        }
+        if (isUsingEmergencyHeat != other.isUsingEmergencyHeat) {
+            return false;
+        }
+        if (lockedTemperatureHigh == null) {
+            if (other.lockedTemperatureHigh != null) {
+                return false;
+            }
+        } else if (!lockedTemperatureHigh.equals(other.lockedTemperatureHigh)) {
+            return false;
+        }
+        if (lockedTemperatureLow == null) {
+            if (other.lockedTemperatureLow != null) {
+                return false;
+            }
+        } else if (!lockedTemperatureLow.equals(other.lockedTemperatureLow)) {
+            return false;
+        }
+        if (mode == null) {
+            if (other.mode != null) {
+                return false;
+            }
+        } else if (!mode.equals(other.mode)) {
+            return false;
+        }
+        if (previousMode == null) {
+            if (other.previousMode != null) {
+                return false;
+            }
+        } else if (!previousMode.equals(other.previousMode)) {
+            return false;
+        }
+        if (state == null) {
+            if (other.state != null) {
+                return false;
+            }
+        } else if (!state.equals(other.state)) {
+            return false;
+        }
+        if (sunlightCorrectionActive != other.sunlightCorrectionActive) {
+            return false;
+        }
+        if (sunlightCorrectionEnabled != other.sunlightCorrectionEnabled) {
+            return false;
+        }
+        if (targetTemperature == null) {
+            if (other.targetTemperature != null) {
+                return false;
+            }
+        } else if (!targetTemperature.equals(other.targetTemperature)) {
+            return false;
+        }
+        if (targetTemperatureHigh == null) {
+            if (other.targetTemperatureHigh != null) {
+                return false;
+            }
+        } else if (!targetTemperatureHigh.equals(other.targetTemperatureHigh)) {
+            return false;
+        }
+        if (targetTemperatureLow == null) {
+            if (other.targetTemperatureLow != null) {
+                return false;
+            }
+        } else if (!targetTemperatureLow.equals(other.targetTemperatureLow)) {
+            return false;
+        }
+        if (tempScale == null) {
+            if (other.tempScale != null) {
+                return false;
+            }
+        } else if (!tempScale.equals(other.tempScale)) {
+            return false;
+        }
+        if (timeToTarget == null) {
+            if (other.timeToTarget != null) {
+                return false;
+            }
+        } else if (!timeToTarget.equals(other.timeToTarget)) {
+            return false;
+        }
+        if (whereName == null) {
+            if (other.whereName != null) {
+                return false;
+            }
+        } else if (!whereName.equals(other.whereName)) {
+            return false;
+        }
+        return true;
```

Comment:
That is some demonic equals method, in intellij I can generate one which uses `Objects.equals` which looks less demonic:

```java
if (this == o)
    return true;
if (o == null || getClass() != o.getClass())
    return false;
ThingManager that = (ThingManager) o;
return Objects.equals(logger, that.logger) && Objects.equals(scheduler, that.scheduler) && Objects
        .equals(eventPublisher, that.eventPublisher) && Objects
        .equals(itemChannelLinkRegistry, that.itemChannelLinkRegistry) && Objects
        .equals(thingHandlerFactories, that.thingHandlerFactories) && Objects
        .equals(thingHandlers, that.thingHandlers) && Objects
        .equals(thingHandlersByFactory, that.thingHandlersByFactory) && Objects
        .equals(thingTypeRegistry, that.thingTypeRegistry) && Objects
        .equals(thingStatusInfoI18nLocalizationService, that.thingStatusInfoI18nLocalizationService) && Objects
        .equals(thingHandlerCallback, that.thingHandlerCallback) && Objects
        .equals(thingRegistry, that.thingRegistry) && Objects
        .equals(configDescriptionRegistry, that.configDescriptionRegistry) && Objects
        .equals(things, that.things) && Objects.equals(registerHandlerLock, that.registerHandlerLock) && Objects
        .equals(thingUpdatedLock, that.thingUpdatedLock) && Objects
        .equals(initializationVetoManager, that.initializationVetoManager);
```